Prolog/models/gameState.pl

```diff
@@ -1,3 +1,33 @@
-:- module(gameState,[testMethodGameState/1]).
-
-testMethodGameState(Param):- write(Param).
\ No newline at end of file
+:- module(gameState, [ create/6, incrementScore/3]).
+
+playingScreenType('playing-screen').
+pausedScreenType('paused-screen').
+gameOverScreenType('game-over-screen').
+
+create(Bird, PipeGroups, Score, HighestScore, ScreenType, GameState) :-
+  GameState=gameState(bird(Bird), pipeGroups(PipeGroups), score(Score), highestScore(HighestScore), screenType(ScreenType)).
+
+bird(GameState, Bird) :-
+  GameState=gameState(bird(Bird), pipeGroups(_), score(_), highestScore(_), screenType(_)).
+
+pipeGroups(GameState, PipeGroups) :-
+  GameState=gameState(bird(_), pipeGroups(PipeGroups), score(_), highestScore(_), screenType(_)).
+
+score(GameState, Score) :-
+  GameState=gameState(bird(_), pipeGroups(_), score(Score), highestScore(_), screenType(_)).
+
+incrementScore(GameState, Score, Increment) :-
+  NewScore is Score+Increment,
+  GameState=gameState(bird(_), pipeGroups(_), score(NewScore), highestScore(_), screenType(_)).
+
+highestScore(GameState, HighestScore) :-
+  GameState=gameState(bird(_), pipeGroups(_), score(_), highestScore(HighestScore), screenType(_)).
+
+screenType(GameState, ScreenType) :-
+  isScreenType(ScreenType),
```

Comment:
Shouldn't this `isScreenType` check be in the creation procedure? If we know the game state was created with a valid screen type, we don't need to check it again here.